commons-hpcc/src/main/java/org/hpccsystems/commons/ecl/FieldDef.java

```diff
@@ -75,38 +66,48 @@ public FieldDef(String fieldName, TypeDef typeDef)
      * @param defs
      *            the array of fields composing this def
      */
-    public FieldDef(String fieldName, FieldType fieldType, String typeName, long len, boolean isFixedLength, HpccSrcType styp, FieldDef[] defs)
+    public FieldDef(String fieldName, FieldType fieldType, String typeName, long len, boolean isFixedLength,
+            boolean isUnsigned, HpccSrcType styp, FieldDef[] defs)
```

Comment:
It was a valid concern in that the code could not handle it. However, HPCC does support lengths greater than the Java Integer.MAX_VALUE and it will also use a value larger than this for variable length fields. So, I updated the code base to handle these cases correctly